### PR TITLE
Add UpdatePerfProfile endpoint

### PR DESCRIPTION
**Pull request by khansaad**
## Description

This PR emables the updatePerformanceProfile endpoint to let the users upgrade the performance profile to newer versions.
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

Please describe the tests that were run to verify your changes and steps to reproduce. Please specify any test configuration required. 

- [ ] New Test X
- [ ] Functional testsuite

**Test Configuration**
* Kubernetes clusters tested on: 

## Checklist :dart:

- [ ] Followed coding guidelines
- [ ] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [ ] Tests added or updated

## Additional information

Test Image: `quay.io/khansaad/autotune_operator:update-profile`

### CODE REVIEW

File: manifests/crc/default-db-included-installation/openshift/kruize-crc-openshift.yaml [hunk 1]
```diff
@@ -215,6 +215,7 @@ data:
       "experimentNameFormat" : "%datasource%|%clustername%|%namespace%|%workloadname%(%workloadtype%)|%containername%",
       "bulkapilimit" : 1000,
       "isKafkaEnabled" : "false",
+      "perfProfileSupportedVersion" : 2.0,
```
Comment (chandrams) [inline manifests/crc/default-db-included-installation/openshift/kruize-crc-openshift.yaml]: Please update the API doc

File: src/main/java/com/autotune/analyzer/services/PerformanceProfileService.java [hunk 2]
```diff
@@ -159,17 +160,84 @@ public boolean shouldSkipClass(Class<?> aClass) {
...
+            }
+            // Compare version — only allow update if the version is matching with the current supported one
+            if (incomingPerfProfile.getProfile_version() != KruizeDeploymentInfo.perf_profile_supported_version) {
+                LOGGER.debug("Version not supported");
+                sendErrorResponse(
+                        response,
+                        null,
+                        HttpServletResponse.SC_CONFLICT,
+                        String.format(
+                                "Update rejected: the provided version (%.1f) is older than the current version (%.1f) for profile '%s'.",
```
Comment (chandrams) [inline src/main/java/com/autotune/analyzer/services/PerformanceProfileService.java]: Should the message say "profile version does not match the supported profile version" rather than "older than the current version"?

File: manifests/crc/default-db-included-installation/openshift/kruize-crc-openshift.yaml (hunk 1 above)
Comment (khansaad) [reply manifests/crc/default-db-included-installation/openshift/kruize-crc-openshift.yaml]: Done

File: src/main/java/com/autotune/analyzer/services/PerformanceProfileService.java (hunk 2 above)
Comment (khansaad) [reply src/main/java/com/autotune/analyzer/services/PerformanceProfileService.java]: Makes sense. Updated now

File: src/main/java/com/autotune/analyzer/serviceObjects/Converters.java [hunk 3]
```diff
@@ -393,40 +393,54 @@ public static ExperimentResultData convertUpdateResultsAPIObjToExperimentResultD
...
         }
 
-        public static PerformanceProfile convertInputJSONToCreatePerfProfile(String inputData) throws InvalidValueException {
+        public static PerformanceProfile convertInputJSONToCreatePerfProfile(String inputData) throws InvalidValueException, Exception {
             PerformanceProfile performanceProfile = null;
             if (inputData != null) {
                 JSONObject jsonObject = new JSONObject(inputData);
                 String perfProfileName = jsonObject.getString(AnalyzerConstants.AutotuneObjectConstants.NAME);
-                Double profileVersion = jsonObject.has(AnalyzerConstants.PROFILE_VERSION) ? jsonObject.getDouble(AnalyzerConstants.PROFILE_VERSION) : null;
+                double profileVersion = jsonObject.has(AnalyzerConstants.PROFILE_VERSION) ? jsonObject.getDouble(AnalyzerConstants.PROFILE_VERSION) : 0.0;
```
Comment (bharathappali) [inline src/main/java/com/autotune/analyzer/serviceObjects/Converters.java]: would it be a good idea to make `0.0` as a constant?

**Review by bharathappali — CHANGES_REQUESTED** [review]
Apart from the constants [which is a trivial one]. everything else LGTM

File: src/main/java/com/autotune/analyzer/serviceObjects/Converters.java (hunk 3 above)
Comment (khansaad) [reply src/main/java/com/autotune/analyzer/serviceObjects/Converters.java]: Done

File: design/PerformanceProfile.md [hunk 4]
```diff
@@ -96,52 +96,71 @@ Same as in the createPerformanceProfile. You can refer [here](/design/Performanc
...
+    * Duplicate API request.
+    ```
+    {
+        "message": "Validation failed: Performance profile <name> already updated with the version <version>",
+        "httpcode": 409,
+        "documentationLink": "",
+        "status": "ERROR"
+    }
+    ```
+    * Duplicate SLO request.
```
Comment (chandrams) [inline design/PerformanceProfile.md:139]: Please include tests for Duplicate SLO request & superset error messages

Comment (khansaad) [reply design/PerformanceProfile.md:139]: Added as part of this [PR](https://github.com/kruize/autotune/pull/1663)

**Review by dinogun — APPROVED** [review]
LGTM
